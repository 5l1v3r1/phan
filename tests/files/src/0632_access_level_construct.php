<?php

class A632 {
    public function __construct() {}
}

class B632 extends A632 {
    protected function __construct() {
        parent::__construct();
    }
}

abstract class AbstractClass632 {
    abstract public function __construct();
}

class ChildClass632 extends AbstractClass632 {
    protected function __construct() {}
}
