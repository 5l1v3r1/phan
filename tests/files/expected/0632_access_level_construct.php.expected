%s:8 PhanConstructAccessSignatureMismatch Access level to function __construct() must be compatible with function __construct() : void defined in %s:4
%s:18 PhanConstructAccessSignatureMismatch Access level to function __construct() must be compatible with function __construct() : void defined in %s:14
